includes/admin/class-wp-job-manager-admin-notices.php

```diff
@@ -144,13 +144,7 @@ public static function has_notice( $notice ) {
 	public static function init_core_notices() {
 		// core_setup: Notice is used when first activating WP Job Manager.
 		add_action( 'job_manager_admin_notice_' . self::NOTICE_CORE_SETUP, [ __CLASS__, 'display_core_setup' ] );
-		add_action(
-			'job_manager_admin_notice_' . self::NOTICE_ADDON_UPDATE_AVAILABLE,
-			[
-				__CLASS__,
-				'display_addon_update_available',
-			]
-		);
+		add_filter( 'wpjm_admin_notices', [ __CLASS__, 'maybe_add_addon_update_available_notice' ], 10, 1 );
```

Comment:
I think this should be moved up into `init`. I think `init_core_notices` is only loaded when the legacy system is and the legacy notices aren't loaded on Ajax calls (for the dismiss action, for example). This is making it so the dismiss action doesn't work because it isn't a known notice when it gets to `handle_notice_dismiss`.